feat: Add API Routes and added axios version 1.6.7

sidarth-23 opened the pull request:
<!--
  Thanks for submitting a pull request!
  We appreciate you spending the time to work on these changes. Please provide enough information so that others can review your pull request. 
-->

## Summary
- Add types for login success and error responses. 
- Added BASE_URL to use environment variable or default to 'http://127.0.0.1:8000/api/'

<!--
 Explain the **motivation** of this Pull Request
-->

- [x] Tested (Must)
- [ ] Test Case added
- [x] Build Successful (Must)
- [x] Sufficient Code comments added (Must)
- [ ] Attached Screenshots / Videos <!-- if the PR contains UI changed, fixes, improvements -->
- [x] All Relevant Documents added


## Depends on

<!--- Does this PR depend on another PR that should be merged first or at the same time -->

yaju09 commented on apps/admin/api-routes/base-url.ts:
```diff
@@ -0,0 +1 @@
+export const BASE_URL = process.env.BASE_URL || 'http://127.0.0.1:8000/api/';
```
```suggestion
export const BASE_URL = process.env.BASE_URL || 'http://127.0.0.1:8000/api/';
```
will it be better to have diff base url's for diff env's ?

BASE_URL_LOCAL
BASE_URL_STAGING
BASE_URL_PROD

yogai53 replied:
@yaju09 we don't have any plan for infrastructure. So let's not plan for environment related works.

sidarth-23 replied:
@yogai53 Then should I hard code the url for now ?

yaju09 replied:
@yogai53 got it Yoga

yaju09 replied:
@sidarth-23 the above code is fine in that case.

sidarth-23 committed:
feat: Add axios version 1.6. 

- Add types for login success and error responses. 
- Added BASE_URL to use environment variable or default to 'http://127.0.0.1:8000/api/'
fix: remove default fallback value for BASE_URL to rely solely on environment variable

sidarth-23 committed:
Merge branch 'master' into feat/add-base-url-and-authentication-files
